Libraries/LibGUI/FileSystemModel.cpp

```diff
@@ -73,13 +74,11 @@ bool FileSystemModel::Node::fetch_data(const String& full_path, bool is_root)
     mtime = st.st_mtime;
 
     if (S_ISLNK(mode)) {
-        char buffer[PATH_MAX];
-        int length = readlink(full_path.characters(), buffer, sizeof(buffer));
-        if (length < 0) {
+        String link = Core::File::read_link(full_path.characters());
+        if (link.is_null()) {
             perror("readlink");
```

Comment:
In general, we can't rely on `errno` still having a relevant value like this, since any number of things may overwrite it (`malloc` is a common one.)
Perhaps it's still fine in this case, but I would prefer that we not use things like `errno` or `perror()` except directly after calls to LibC that communicate errors via `errno`.
If you want to log the error from `readlink()`, I think it would be better to do that inside `Core::File::read_link()`.